src/pybind/mgr/progress/module.py

```diff
@@ -368,7 +364,64 @@ def _osd_out(self, old_map, old_dump, new_map, osd_id):
         ev.pg_update(self.get("pg_dump"), self.log)
         self._events[ev.id] = ev
 
-    def _osd_in(self, osd_id):
+    def _osd_in(self, old_map, old_dump, new_map, osd_id):
+        affected_pgs = []
+        unmoved_pgs = []
+        for pool in old_dump['pools']:
+            pool_id = pool['pool']
+            for ps in range(0, pool['pg_num']):
+                up_acting = old_map.pg_to_up_acting_osds(pool['pool'], ps)
```

Comment:
for an osd being marked in, being in the new acting set, but not the old acting set, would be expected